backend/src/routes/upload.ts

```diff
@@ -169,11 +204,14 @@ export class UploadRoutes {
       next({ status: 422, error: 'Request is missing measurementDate or measurementDate is invalid'})
       return
     }
-    if (!('instrument' in body) || !body.instrument) {
-      next({ status: 422, error: 'Request is missing instrument'})
+    if (!(('instrument' in body && body.instrument) || ('model' in body && body.model))) {
+      next({ status: 422, error: 'Request is missing either instrument or model'})
```

Comment:
Sanottaisin tämän eri tavalla, esim. Request must have either the field 'instrument' or the field 'model'. Niin, etteivät sitet hämmentyisi turhaan.